nonos/main.py

```diff
@@ -680,503 +737,201 @@ def plot(
-        except IndexError:
-            return None
-
-    def euler(self, vx, vy, x, y, reverse):
-        """
-        Euler integrator for computing the streamlines.
-        Parameters:
-        ----------
-
-        x,y: Floats.
-             Initial condition
-        reverse: Boolean.
-                 If reverse is true, the integration step is negative.
-
-        Output
-        ------
-
-        (dx,dy): (float,float).
-                 Are the azimutal and radial increments.
-                 Only works for cylindrical coordinates.
-        """
-        sign = 1.0
-        if reverse:
-            sign = -1
-        vr = self.get_v(vx, x, y)
-        vt = self.get_v(vy, x, y)
-        if None in (vt, vr):
-            # Avoiding problems...
-            return None, None
-
-        l = np.min(
-            (
-                ((self.x.max() - self.x.min()) / self.nx),
-                ((self.y.max() - self.y.min()) / self.ny),
+        coordgrid = np.array(coordgrid, dtype=object)[np.array(plane) - 1]
+
+        # (coord[0]=R,coord[1]=phi by default)
+        # then, depending on the geometry,
+        # we transform these 2D coordinates arrays
+        transform = coordgrid
+        if func_proj is not None:
+            transform = func_proj(
+                coordgrid[0],
+                coordgrid[1],
+                coordgrid[2],
             )
-        )
-        h = 0.5 * l / np.sqrt(vr ** 2 + vt ** 2)
-
-        return sign * h * np.array([vr, vt / x])
-
-    def get_stream(
-        self,
-        vx,
-        vy,
-        x0,
-        y0,
-        nmax=1000000,
-        maxlength=4 * np.pi,
-        bidirectional=True,
-        reverse=False,
-    ):
-        """
-        Function for computing a streamline.
-        Parameters:
-        -----------
-
-        x0,y0: Floats.
-              Initial position for the stream
-        nmax: Integer.
-              Maxium number of iterations for the stream.
-        maxlength: Float
-                   Maxium allowed length for a stream
-        bidirectional=True
-                      If it's True, the stream will be forward and backward computed.
-        reverse=False
-                The sign of the stream. You can change it mannualy for a single stream,
-                but in practice, it's recommeneded to use this function without set reverse
-                and setting bidirectional = True.
-
-        Output:
-        -------
-
-        If bidirectional is False, the function returns a single array, containing the streamline:
-        The format is:
-
-                                          np.array([[x],[y]])
 
-        If bidirectional is True, the function returns a tuple of two arrays, each one with the same
-        format as bidirectional=False.
-        The format in this case is:
-
-                                (np.array([[x],[y]]),np.array([[x],[y]]))
+        if is_set(lic) and self.code != "fargo3d":
```

Comment:
Although currently equivalent, it's much safer and readable to check for the cases we want rather than the one we don't.
```suggestion
        if is_set(lic) and self.code in ("pluto", "idefix"):
```